pyobis/occurrences/occurrences.py

```diff
@@ -54,48 +40,50 @@ def search(scientificname=None,
         # for the two queries below.
-        occ.search(scientificname = 'Mola mola', offset=0, limit=10)
-        occ.search(scientificname = 'Mola mola', offset=10, limit=10)
+        occ.search(scientificname = 'Mola mola', offset=0, size=10)
+        occ.search(scientificname = 'Mola mola', offset=10, size=10)
 
         # Search on a bounding box
         ## in well known text format
         occ.search(geometry='POLYGON((30.1 10.1, 10 20, 20 40, 40 40, 30.1 10.1))', limit=20)
         from pyobis import taxa
         res = taxa.search(scientificname='Mola mola')['results'][0]
-        occ.search(obisid=res['id'], geometry='POLYGON((30.1 10.1, 10 20, 20 40, 40 40, 30.1 10.1))', limit=20)
-        occ.search(aphiaid=res['worms_id'], geometry='POLYGON((30.1 10.1, 10 20, 20 40, 40 40, 30.1 10.1))', limit=20)
-
-        # Get occurrences for a particular eventDate
-        occ.search(aphiaid=res['worms_id'], year="2013", limit=20)
+        occ.search(obisid=res['id'], geometry='POLYGON((30.1 10.1, 10 20, 20 40, 40 40, 30.1 10.1))', size=20)
+        occ.search(aphiaid=res['worms_id'], geometry='POLYGON((30.1 10.1, 10 20, 20 40, 40 40, 30.1 10.1))', size=20)
 
         # Get mof response as list of pandas dataframes
         occ.search(scientificname="Abra",mof=True,hasextensions="MeasurementOrFact")
     '''
     url = obis_baseurl + 'occurrence'
     scientificname = handle_arrstr(scientificname)
-    out = obis_GET(url, {
-            'taxonid': taxonid,
-            'obisid': obisid,
-            'datasetid': datasetid,
-            'scientificname': scientificname,
-            'startdate': startdate,
-            'enddate': enddate,
-            'startdepth': startdepth,
-            'enddepth': enddepth,
-            'geometry': geometry,
-            'year': year,
-            'fields': fields,
-            'flags': flags,
-            'limit': limit,
-            'offset': offset,
-            'mof': mof,
+    args = {
+            'taxonid': taxonid,'nodeid': nodeid,'datasetid': datasetid,
+            'scientificname': scientificname,'startdate': startdate,
+            'enddate': enddate,'startdepth': startdepth,'enddepth': enddepth,
+            'geometry': geometry,'year': year,'fields': fields,
+            'flags': flags,'offset': offset,'mof': mof,'size':0,
             'hasextensions': hasextensions
-        }, 'application/json; charset=utf-8', **kwargs)
-    if (mof):
-        mofNormalized = pd.json_normalize(out["results"], "mof", ["scientificName","id", "eventDate"])
-        a = pd.merge(pd.DataFrame(out["results"]),mofNormalized,on='id',how='inner')
-        ids = a.id.unique()
-        return [a[a['id']==ids[i]] for i in range(len(ids))]
+        }
+    out  = obis_GET(url,args,'application/json; charset=utf-8', **kwargs)
+    limit = out["total"] if size == 5000 else size
+    for i in range(5000,limit+1,5000):
+        if args["size"]!=0:
```

Comment:
Actually, this condition is to make sure that we fetch the first page (or batch) of data correctly.

Initially, the `size` parameter is set to 0, so that we get only the total number of records fast without eating up memory. Then we get into the `for` loop and fetch the first batch of records (or the first page). After we are done fetching the first batch, we change the `size` parameter to 5000 i.e., the default value and insert the `after` parameter to fetch only subsequent records. 

This `if` condition helps us determine whether we are fetching the first batch or subsequent batches, if the latter is true then we also insert the `after` parameter in our request.

I will add some comments to the code as per your suggestion.